thing/index.js

```diff
@@ -514,9 +549,20 @@ function ThingShadowsClient(deviceOptions, thingShadowOptions) {
          // which the application will need to filter out.
          //
          if (thingShadows[thingName].persistentSubscribe === true) {
-            this._handleSubscriptions(thingName, ['update', 'get', 'delete'], ['accepted', 'rejected'],
-               'subscribe');
+            topicsObject.push({
+               operations: ['update', 'get', 'delete'],
+               statii: ['accepted', 'rejected'],
+            });
+         }
+
+         if (topicsObject.length > 0) {
+            this._handleSubscriptions(thingName, topicsObject, 'subscribe', callback);
```

Comment:
Are you think about moving the logic for checking if all topics are granted into the callback? How are we going to handle all the other places that uses the _handleSubscriptions to sub?